nifi-nar-bundles/nifi-iotdb-bundle/nifi-iotdb-processors/src/test/java/org/apache/nifi/processors/AbstractIoTDBUT.java

```diff
@@ -0,0 +1,282 @@
+        assertEquals(exceptedMsg, result.getValue());
+
+        // schema with wrong key
+        schemaAttribute =
+                "{\n"
+                        + "\t\"timeName\": \"Time\",\n"
+                        + "\t\"fields\": [{\n"
+                        + "\t\t\"tsName\": \"s1\",\n"
+                        + "\t\t\"dataType\": \"INT32\",\n"
+                        + "\t\t\"encode\": \"RLE\"\n"
+                        + "\t}, {\n"
+                        + "\t\t\"tsName\": \"s2\",\n"
+                        + "\t\t\"dataType\": \"DOUBLE\",\n"
+                        + "\t\t\"encoding\": \"PLAIN\"\n"
+                        + "\t}]\n"
+                        + "}";
+
+        result = processor.validateSchemaAttribute(schemaAttribute);
+        exceptedMsg = "Unknown property or properties: [encode]";
+
+        assertEquals(false, result.getKey());
+        assertEquals(exceptedMsg, result.getValue());
+
+        // schema with wrong compression type
+        schemaAttribute =
+                "{\n"
+                        + "\t\"timeName\": \"Time\",\n"
+                        + "\t\"fields\": [{\n"
+                        + "\t\t\"tsName\": \"s1\",\n"
+                        + "\t\t\"dataType\": \"INT32\",\n"
+                        + "\t\t\"encoding\": \"RLE\",\n"
+                        + "\t\t\"compressionType\": \"ZIP\"\n"
+                        + "\t}, {\n"
+                        + "\t\t\"tsName\": \"s2\",\n"
+                        + "\t\t\"dataType\": \"DOUBLE\",\n"
+                        + "\t\t\"encoding\": \"PLAIN\",\n"
+                        + "\t\t\"compressionType\": \"GZIP\"\n"
+                        + "\t}]\n"
+                        + "}";
+
+        result = processor.validateSchemaAttribute(schemaAttribute);
+        exceptedMsg =
+                "Unknown `compressionType`: ZIP, The supported compressionType are [LZO, PAA, SDT, UNCOMPRESSED, PLA, LZ4, GZIP, SNAPPY]";
+
+        assertEquals(false, result.getKey());
+        assertEquals(exceptedMsg, result.getValue());
+    }
+
+    @Test
+    public void testParseSchema() {
+        ArrayList<String> filedNames =
+                new ArrayList<String>() {
+                    {
+                        add("root.sg1.d1.s1");
+                        add("root.sg1.d1.s2");
+                        add("root.sg1.d2.s1");
+                    }
+                };
+        Map<String, List<String>> deviceMeasurementMap = processor.parseSchema(filedNames);
+        HashMap<String, List<String>> exceptedMap =
+                new HashMap<String, List<String>>() {
+                    {
+                        put(
+                                "root.sg1.d1",
+                                new ArrayList<String>() {
+                                    {
+                                        add("s1");
+                                        add("s2");
+                                    }
+                                });
```

Comment:
This kind of anonymous instantiation should not be used. Instead, Arrays.asList() should be used, please update all references.

Reply:
Resolved